src/taco-utils/promisesUtils.ts

```diff
@@ -0,0 +1,31 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/lodash.d.ts" />
+
+"use strict";
+
+import _ = require("lodash");
+
+module TacoUtility {
+    export class PromisesUtils {
+        public static executeAfter<T>(valueOrPromise: T, afterCallback: { (): void }, failCallback?: { (reason: any): void }): T {
```

Comment:
I'm confused by what this function does. How is it different to Q(valueOrPromise).finally(afterCallback).fail(failCallback) ?